core/src/main/java/feast/core/util/TypeConversion.java

```diff
@@ -67,12 +67,10 @@
    * Marshals a given map into its corresponding json string
    *
    * @param map
-   * @return json string corresponding to given map
+   * @return json string corresponding to given map or null if the map is empty
    */
   public static String convertMapToJsonString(Map<String, String> map) {
```

Comment:
I let myself change this, because it aligns better with use cases inside the project. Is it likely that this is being used outside?

Reply:
This method seems not be used at all in the current project except being referenced its unit test.
I'm for not returning a `null` as its not semantic:
- a convert to JSON method should return valid, parsable JSON.
- `convertMapToJsonString(map) == null` makes less sense than `map.isEmpty()`

We should not be using `null` unless necessary. What is wrong with just using `map.isEmpty()`?

Reply:
> Is it likely that this is being used outside?

For the general side of this, I think it's safe to say nothing in `core`, `ingestion` and `serving` modules is user API with expected stability, only the SDKs need to be given the extra considerations appropriate for library code.

On this specific change, I agree with with all of @mrzzy's points.

Reply:
> I think it's safe to say nothing in `core`, `ingestion` and `serving` modules is user API with expected stability, only the SDKs need to be given the extra considerations appropriate for library code.

Well, with the recent introduction of https://github.com/gojek/feast/tree/master/storage/api "only the SDKs" should be extended to the storage APIs too probably.

Reply:
I got your points. We're now storing empty json for features with an empty labels map.